vimdoc/module.py

```diff
@@ -141,25 +142,35 @@ def Close(self):
     if ((self.GetCollection(vimdoc.FLAG) or
          self.GetCollection(vimdoc.SETTING)) and
         'config' not in self.sections):
-      config = Block()
-      config.SetType(vimdoc.SECTION)
+      config = Block(vimdoc.SECTION)
       config.Local(id='config', name='Configuration')
       self.Merge(config)
-    if not self.order:
-      self.order = []
-      for builtin in [
-          'intro',
-          'config',
-          'commands',
-          'autocmds',
-          'settings',
-          'dicts',
-          'functions',
-          'exceptions',
-          'mappings',
-          'about']:
-        if builtin in self.sections or builtin in self.backmatters:
-          self.order.append(builtin)
+
+    # Use explicit order as partial ordering and merge with default section
+    # ordering. All custom sections must be ordered explicitly.
+    # Custom sections will be ordered after all sections besides 'about'.
+    order = self.order or []
+    builtins_pre = []
+    for builtin in [
+        'intro',
+        'config',
+        'commands',
+        'autocmds',
+        'settings',
+        'dicts',
+        'functions',
+        'exceptions',
+        'mappings']:
+      if builtin in self.sections or builtin in self.backmatters:
+        if builtin not in order:
+          builtins_pre.append(builtin)
+    order = builtins_pre + order
```

Comment:
Isn't this going to run into trouble if `@order` _does_ contain one of the builtins? (i.e. it'll end up in `order` twice).

Some tests for this part would be nice, if possible.